### DISCUSSION TITLE
Fix camelCase name in Mojo stdlib

### DISCUSSION THREAD

**Discussion by gryznar** [modular/modular#384] [Ideas]
Mojo in Testing library has names in camelCase, as Python in unittest. I do not catch arguments on that and I disagree with them. 
1. It is incostinent with PEP8
2. This is not the case of compatibility, because Mojo could have assert_equal and this won't break compatibility with unittest.TestCase::assertEqual()
3. Core Python developers would choose snake_case in unittest if they could do it one more time
4. Only Testing module has camelCase
5. A lot of people from Python community do not care about unittest and use pytest instead
6. If Mojo would have 100% compatibility with Python it can just import unittest / pytest or any other lib and use it instead of stdlib Testing
7. If someone would like to find method for assertion equality he won't search for "assertEqual", but will look into docs / ask someone or search for term "assert" / "fn assert" or similar
8. Idea to make names in Mojo stdlib the same as in Python (if it is not the case of compatibility) can make a headache in maintanance very quickly

**Comment by lattner** [comment]
Yep, this is simply a mistake.  @stumpOS do you know who owns this code?  Can you take a look?

**Reply by ylmzfurkan** [reply]
I understand that you have concerns regarding the use of camelCase naming convention in the Mojo testing library. You mention that it is inconsistent with PEP8 and that you disagree with this decision.

While the choice of naming convention is subjective and may vary between different testing libraries, it is important to note that Mojo may have its own design considerations and principles that led to the adoption of camelCase. It is not necessarily a matter of compatibility with Python's unittest library.

It is true that Python's unittest library follows the snake_case naming convention, and it is the preferred convention in the Python community. However, Mojo, being a separate testing library, has the flexibility to choose its own naming conventions based on its specific goals and design philosophy.

Regarding the argument that people may have difficulty finding assertion methods with camelCase names, it's important to note that most testing libraries provide documentation and search functionality to assist users in finding the appropriate methods. Additionally, using common terms like "assert" or "fn assert" in searches can help users discover the relevant methods.

While it is understandable that you would prefer consistency between Mojo and Python's naming conventions, it's worth considering that making changes to align with Python's naming conventions could introduce maintenance challenges and potential confusion for existing users of Mojo.

Ultimately, the choice of naming conventions in a testing library is a decision made by its developers based on various factors. If you strongly believe that the naming convention in Mojo should be reconsidered, you may consider raising your concerns with the maintainers of the library or engaging in discussions within the Mojo community to express your viewpoint.

**Comment by lattner** [comment]
Mojo is a member of the Python family.  You're right that some naming is subjective, but PEP8 is pretty widely adopted across the Python community - I don't see a reason for Mojo to diverge.

**Reply by ylmzfurkan** [reply]
If Mojo diverges from PEP8 naming conventions without a specific reason or compatibility requirement, it might introduce unnecessary cognitive overhead for developers using the library. By aligning with PEP8, Mojo can leverage the familiarity and established conventions within the Python community.

Considering the importance of consistency and ease of use within the Python ecosystem, it would be beneficial to have a discussion within the Mojo community about the possibility of aligning the naming conventions with PEP8. This can lead to a more unified experience for Python developers using Mojo and promote better collaboration within the community.

Engaging in constructive discussions and providing feedback within the Mojo community can help foster a better understanding of the concerns and considerations around naming conventions, ultimately leading to a decision that best serves the interests of the Python community as a whole.

**Reply by gryznar** [reply]
It is fixed, topic could be closed
